pkg/registry/garden/shoot/strategy.go

```diff
@@ -97,9 +97,16 @@ func mustIncreaseGeneration(oldShoot, newShoot *garden.Shoot) bool {
 				mustIncrease = true
 			}
 		default:
-			// The shoot state is not failed and the reconcile annotation is set.
-			if val, ok := newShoot.Annotations[common.ShootOperation]; ok && val == common.ShootOperationReconcile {
-				mustIncrease = true
+			// The shoot state is not failed and the reconcile or rotate-credentials annotation is set.
+			if val, ok := newShoot.Annotations[common.ShootOperation]; ok {
+				if val == common.ShootOperationReconcile {
+					mustIncrease = true
+				}
+				if val == common.ShootOperationRotateKubeconfigCredentials {
```

Comment:
Can you add documentation for this?

We have lots of annotations and they are very undocumented.

Reply:
Done